hpe3parclient/client.py

```diff
@@ -968,32 +988,44 @@ def copyVolume(self, src_name, dest_name, dest_cpg, optional=None):
         # has to be taken care by caller side
         if optional:
             if self.primera_supported:
-                # For primera array there is no compression and tdvv parameters
-                # removing tdvv and replacing compression with reduce field
-                if 'tdvv' in optional:
-                    optional.pop('tdvv')
-
-                if optional.get('tpvv') is True \
+                for key in ['tpvv', 'compression', 'tdvv']:
+                    option = optional.get(key)
+                    if option and option not in [True, False]:
+                        # raising exception for junk compression input
+                        ex_desc = "39 - invalid input: wrong type for key " \
+                            "[%s], Valid values are[True, False]" % key
+                        raise exceptions.HTTPBadRequest(ex_desc)
```

Comment:
Please put a full stop after [%s].
And also, please put a space before [True, False]

Reply:
done